File: java_solutions/2644.find-the-maximum-divisibility-score.java
/*
 * @lc app=leetcode id=2644 lang=java
 *
 * [2644] Find the Maximum Divisibility Score
 */

class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        return Arrays.stream(divisors)
            .distinct()
            .boxed()
            .max(Comparator.<Integer, Long>comparing(d -> Arrays.stream(nums).filter(n -> n % d == 0).count())
                .thenComparing(Comparator.reverseOrder()))
            .orElse(0);
    }
}

File: kotlin_solutions/2644.find-the-maximum-divisibility-score.kt
/*
 * @lc app=leetcode id=2644 lang=kotlin
 *
 * [2644] Find the Maximum Divisibility Score
 */

class Solution {fun maxDivScore(nums: IntArray, divisors: IntArray) = divisors.sorted().map { div -> nums.count { it % div == 0 } }.let { counts -> divisors.sorted()[counts.indexOf(counts.maxOrNull())] }}
